homeassistant/components/uptimerobot/binary_sensor.py

```diff
@@ -21,69 +32,91 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_API_KEY): cv.string})
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+@dataclass
+class UptimeRobotBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Entity description for UptimeRobotBinarySensor."""
+
+    target: str = ""
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_entities, discovery_info=None
+):
     """Set up the Uptime Robot binary_sensors."""
+    uptime_robot_api = UptimeRobot()
+    api_key = config[CONF_API_KEY]
+
+    async def async_update_data():
+        """Fetch data from API UptimeRobot API."""
+
+        def api_wrapper():
```

Comment:
We could move the wrapper function outside `async_update_data` to avoid defining the function every update.